### PR TITLE
changed blobs to be in arrayBufferLike format instead of string

### LINKED ISSUE #5263: buildSnapshotTree and storages using them do unnecessary base64 encoding / decoding
Basically it boils down to ChannelStorageService & BlobCacheStorageService having extra blobs as ArrayBuffers, not as base64 encoded strings. This pushes the problem up the stack, where we get these base64 strings from buildSnapshotTree (see flattenCore), which converts input (utf8 or base64) to base64. Instead, it should return arrays and let callers deal with it
 extra blobs should be ArrayBuffers, not base64 encoded strings.

There are other usages affected by it, but they all can and should be converted to using arrays, not base64 encoded strings.

### PR DESCRIPTION
**Pull request by chensixx**
Fixes #5263 
Changed extraBlobs in ChannelStorageService and BlobCacheStorageService to be type arrayBufferLike, not base64 strings. And made changes in affected places.

### CODE REVIEW

File: packages/loader/driver-utils/src/buildSnapshotTree.ts [hunk 1]
```diff
@@ -17,23 +17,23 @@ import { v4 as uuid } from "uuid";
...
             const id = uuid();
-            blobMap.set(id, buffer.toString("base64"));
+            blobMap.set(id, buffer);
 
             const entry: git.ITreeEntry = {
                 mode: FileMode[treeEntry.mode],
                 path: subPath,
                 sha: id,
-                size: buffer.length,
+                size: IsoBuffer.from(blob.contents, blob.encoding).length,
```
Comment (vladsud) [inline packages/loader/driver-utils/src/buildSnapshotTree.ts:36]: I'd not do it and write out zero here - it's not used and is not exposed from public function, so no need to waste resources of allocating memory, decoding, etc. - it's a waste that has no value

File: packages/loader/driver-utils/src/test/blobAggregation.spec.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ class InMemoryStorage {
     async readBlob(id: string) {
         const blob = this.blobs.get(id);
         if (blob !== undefined) {
```
Comment (vladsud) [inline packages/loader/driver-utils/src/test/blobAggregation.spec.ts:64]: Not that it matters a lot, but code becomes more streamlined if you do this:
```
assert(blob !== undefined, "unknown blob id");
return blob;
```
here and in all other places like that

**Review by vladsud — APPROVED** [review]
Love the changes! Great work!

File: packages/loader/driver-utils/src/test/blobAggregation.spec.ts (hunk 2 above)
Comment (chensixx) [reply packages/loader/driver-utils/src/test/blobAggregation.spec.ts:64]: i see. it doesn't seem there are other places like this with an error message. other places are more like if blob is undefined do this, else do that

**Pull request merged**